core/lib/src/shutdown.rs

```diff
@@ -0,0 +1,47 @@
+use crate::request::{FromRequest, Outcome, Request};
```

Comment:
This module is quite short (only a single `struct` and two `impls`). If you had to inline this somewhere, where do you think you would choose?

Reply:
Just summing up what I said in IRC/Matrix earlier for anyone else following this: After briefly looking at the other directories and files that exist in the `core/lib`, I honestly don't see anywhere where it would fit sensibly. Though it is comparatively small, the module is significantly different than anything else that currently exists in `core/lib`.